src/docquery/ext/pipeline_image_classification.py

```diff
@@ -0,0 +1,200 @@
+    """
+    Convert a (generated) token sequence into an ordered JSON format.
+    """
+    output = dict()
+
+    while tokens:
+        start_token = re.search(r"<s_(.*?)>", tokens, re.IGNORECASE)
+        if start_token is None:
+            break
+        key = start_token.group(1)
+        end_token = re.search(rf"</s_{key}>", tokens, re.IGNORECASE)
+        start_token = start_token.group()
+        if end_token is None:
+            tokens = tokens.replace(start_token, "")
+        else:
+            end_token = end_token.group()
+            start_token_escaped = re.escape(start_token)
+            end_token_escaped = re.escape(end_token)
+            content = re.search(f"{start_token_escaped}(.*?){end_token_escaped}", tokens, re.IGNORECASE)
+            if content is not None:
+                content = content.group(1).strip()
+                if r"<s_" in content and r"</s_" in content:  # non-leaf node
+                    value = donut_token2json(tokenizer, content, is_inner_value=True)
+                    if value:
+                        if len(value) == 1:
+                            value = value[0]
+                        output[key] = value
+                else:  # leaf nodes
+                    output[key] = []
+                    for leaf in content.split(r"<sep/>"):
+                        leaf = leaf.strip()
+                        if leaf in tokenizer.get_added_vocab() and leaf[0] == "<" and leaf[-2:] == "/>":
+                            leaf = leaf[1:-2]  # for categorical special tokens
+                        output[key].append(leaf)
+                    if len(output[key]) == 1:
+                        output[key] = output[key][0]
+
+            tokens = tokens[tokens.find(end_token) + len(end_token) :].strip()
+            if tokens[:6] == r"<sep/>":  # non-leaf nodes
+                return [output] + donut_token2json(tokenizer, tokens[6:], is_inner_value=True)
+
+    if len(output):
+        return [output] if is_inner_value else output
+    else:
+        return [] if is_inner_value else {"text_sequence": tokens}
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class ImageClassificationPipeline(Pipeline):
+    """
+    Image classification pipeline using any `AutoModelForImageClassification`. This pipeline predicts the class of an
+    image.
+
+    This image classification pipeline can currently be loaded from [`pipeline`] using the following task identifier:
+    `"image-classification"`.
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?filter=image-classification).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        requires_backends(self, "vision")
+        self.check_model_type(
+            TF_MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING
+            if self.framework == "tf"
+            else MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING
+        )
```

Comment:
Could replace everything up to this line in `__init__` with `super().__init__(*args, **kwargs)`

Reply:
commented separately about why we copy/pasted.